netbsd/Platform.c

```diff
@@ -0,0 +1,324 @@
+   { .name = "46 SIGRTMIN+13", .number = 46 },
+   { .name = "47 SIGRTMIN+14", .number = 47 },
+   { .name = "48 SIGRTMIN+15", .number = 48 },
+   { .name = "49 SIGRTMIN+16", .number = 49 },
+   { .name = "50 SIGRTMIN+17", .number = 50 },
+   { .name = "51 SIGRTMIN+18", .number = 51 },
+   { .name = "52 SIGRTMIN+19", .number = 52 },
+   { .name = "53 SIGRTMIN+20", .number = 53 },
+   { .name = "54 SIGRTMIN+21", .number = 54 },
+   { .name = "55 SIGRTMIN+22", .number = 55 },
+   { .name = "56 SIGRTMIN+23", .number = 56 },
+   { .name = "57 SIGRTMIN+24", .number = 57 },
+   { .name = "58 SIGRTMIN+25", .number = 58 },
+   { .name = "59 SIGRTMIN+26", .number = 59 },
+   { .name = "60 SIGRTMIN+27", .number = 60 },
+   { .name = "61 SIGRTMIN+28", .number = 61 },
+   { .name = "62 SIGRTMIN+29", .number = 62 },
+   { .name = "63 SIGRTMAX",    .number = 63 },
+};
+
+const unsigned int Platform_numberOfSignals = ARRAYSIZE(Platform_signals);
+
+const MeterClass* const Platform_meterTypes[] = {
+   &CPUMeter_class,
+   &ClockMeter_class,
+   &DateMeter_class,
+   &DateTimeMeter_class,
+   &LoadAverageMeter_class,
+   &LoadMeter_class,
+   &MemoryMeter_class,
+   &SwapMeter_class,
+   &TasksMeter_class,
+   &UptimeMeter_class,
+   &BatteryMeter_class,
+   &HostnameMeter_class,
+   &SysArchMeter_class,
+   &AllCPUsMeter_class,
+   &AllCPUs2Meter_class,
+   &AllCPUs4Meter_class,
+   &AllCPUs8Meter_class,
+   &LeftCPUsMeter_class,
+   &RightCPUsMeter_class,
+   &LeftCPUs2Meter_class,
+   &RightCPUs2Meter_class,
+   &LeftCPUs4Meter_class,
+   &RightCPUs4Meter_class,
+   &LeftCPUs8Meter_class,
+   &RightCPUs8Meter_class,
+   &BlankMeter_class,
+   NULL
+};
+
+void Platform_init(void) {
+   /* no platform-specific setup needed */
+}
+
+void Platform_done(void) {
+   /* no platform-specific cleanup needed */
+}
+
+void Platform_setBindings(Htop_Action* keys) {
+   /* no platform-specific key bindings */
+   (void) keys;
+}
+
+int Platform_getUptime() {
+   struct timeval bootTime, currTime;
+   const int mib[2] = { CTL_KERN, KERN_BOOTTIME };
+   size_t size = sizeof(bootTime);
+
+   int err = sysctl(mib, 2, &bootTime, &size, NULL, 0);
+   if (err) {
+      return -1;
+   }
+   gettimeofday(&currTime, NULL);
+
+   return (int) difftime(currTime.tv_sec, bootTime.tv_sec);
+}
+
+void Platform_getLoadAverage(double* one, double* five, double* fifteen) {
+   struct loadavg loadAverage;
+   const int mib[2] = { CTL_VM, VM_LOADAVG };
+   size_t size = sizeof(loadAverage);
+
+   int err = sysctl(mib, 2, &loadAverage, &size, NULL, 0);
+   if (err) {
+      *one = 0;
+      *five = 0;
+      *fifteen = 0;
+      return;
+   }
+   *one     = (double) loadAverage.ldavg[0] / loadAverage.fscale;
+   *five    = (double) loadAverage.ldavg[1] / loadAverage.fscale;
+   *fifteen = (double) loadAverage.ldavg[2] / loadAverage.fscale;
+}
+
+int Platform_getMaxPid() {
+   // this is hard-coded in sys/sys/proc.h - no sysctl exists
+   return 30000;
```

Comment:
30000 is only the limit if there are <16K processes. Likely should be sizeof(int32_t), since that's the limit of the pid_t data type?

Reply:
It seem you are right, I shall look into this and get back.

Reply:
This should be fixed in f789aa5